### PR TITLE
sundials: add 5.0.0-dev.1 and minor cleanup

### PR DESCRIPTION
**Pull request by balos1**
CC: @gardner48, @balay

### CODE REVIEW

File: var/spack/repos/builtin/packages/sundials/package.py [hunk 1]
```diff
@@ -130,17 +133,15 @@ class Sundials(CMakePackage):
     # External libraries incompatible with single precision
     conflicts('+klu',   when='precision=single')
     conflicts('+hypre', when='+hypre@:2.12.0 precision=single')
+    conflicts('+superlu-dist', when='precision=single')
 
     # External libraries incompatible with extended (quad) precision
     conflicts('+lapack',     when='precision=extended')
     conflicts('+superlu-mt', when='precision=extended')
+    conflicts('+superlu-dist', when='precision=extended')
```
Comment (gardner48) [inline var/spack/repos/builtin/packages/sundials/package.py]: Add a conflict for `+superlu-dist` with versions before 5.

Comment (gardner48) [inline var/spack/repos/builtin/packages/sundials/package.py]: Need to set the the CMake variables to build the SuperLU_DIST module.

Comment (balos1) [reply var/spack/repos/builtin/packages/sundials/package.py]: Took care of this in the depends_on now. We actually require 6.1.1 or higher.

Comment (balos1) [reply var/spack/repos/builtin/packages/sundials/package.py]: Oops, good catch. Should be taken care of now.

Comment (gardner48) [reply var/spack/repos/builtin/packages/sundials/package.py]: Sorry I meant that `+superlu-dist` should conflict with sundials versions earlier than 5. I'd missed the SuperLU_DIST version requirement, good catch on that one.

File: var/spack/repos/builtin/packages/sundials/package.py [hunk 2]
```diff
@@ -298,6 +296,22 @@ def on_off(varstr):
...
+                '-DSUPERLUDIST_INCLUDE_DIR=%s'
+                % spec['superlu-dist'].prefix.include,
+                '-DSUPERLUDIST_LIBRARY_DIR=%s'
+                % spec['superlu-dist'].prefix.lib,
+                '-DSUPERLUDIST_LIBRARIES=%s'
+                % spec['blas'].libs
+            ])
+            if spec.satisfies('+openmp'):
+                args.extend([
+                '-DSUPERLUDIST_OpenMP=ON'
```
Comment (gardner48) [inline var/spack/repos/builtin/packages/sundials/package.py]: Has the SuperLU_DIST spack package been updated in have a `+openmp` variant?

Comment (balos1) [reply var/spack/repos/builtin/packages/sundials/package.py]: No it has not, but it builds with OpenMP by default.

**Comment by balos1** [comment]
Going to keep this on the balay/xsdk-0.5.0 branch for now.

**Pull request closed without merging**